src/core/Checkbox/Checkbox.test.tsx

```diff
@@ -192,3 +200,32 @@ it('renders correctly with visibility checkbox', () => {
 
   expect(container.querySelector('.iui-checkbox-visibility')).toBeTruthy();
 });
+
+it.each(['', 'not'] as const)(
+  'should %s stop propagation correctly if %s used with label',
+  (labelPresent) => {
+    const wrapperOnClick = jest.fn();
+    const checkboxOnChange = jest.fn();
+    const { container } = render(
+      <div onClick={wrapperOnClick}>
+        <Checkbox
+          label={labelPresent && 'label'}
+          className='my-checkbox'
+          onClick={(e) => e.stopPropagation()}
+          onChange={checkboxOnChange}
+        />
+      </div>,
+    );
+    const checkboxComponent = container.querySelector(
+      '.my-checkbox',
+    ) as HTMLElement;
+    fireEvent.click(checkboxComponent);
+
+    expect(checkboxOnChange).toBeCalled();
+    if (labelPresent) {
+      expect(wrapperOnClick).toBeCalled();
```

Comment:
Is it really ok that it still leaks click event when used with label?

Reply:
I don't think we could really fix it without `labelProps`. But the user now has the ability to wrap unlabeled checkbox with their own `<label>`. Also there is that workaround with `ref` mentioned in #533